codecparsers/bitReader.h

```diff
@@ -73,6 +82,27 @@ class BitReader {
     inline void reload();
 };
 
+template <class T>
+bool BitReader::readT(T& v, uint32_t nbits)
+{
+    uint32_t tmp;
+    if (!read(tmp, nbits))
+        return false;
+    v = tmp;
+    return true;
+}
+
+template <class T>
+bool BitReader::readT(T& v)
+{
+    return readT(v, sizeof(v) << 3);
+}
+
+bool BitReader::readT(bool& v)
+{
+    return readT(v, 1);
+}
+
```

Comment:
The main benefit of readT is it will guess how many bits should read base on the type of `v`. if `v `is `uint32_t `it will read 32 bits, if it’s `bool `it will read 1 bits.
I see peek code in mpeg2.
>bitsToParse = kVLCTable[i][0];
>bitReaderPeekBits(bitsToParse, &mbVLC);

Seems it will read bitsToParse bits. Therefore, you cannot write a template based peek for this. Same thing applied to skip too.
You may need change skip to `bool skip(uint32_t&v, uint32_t bit)`, so you can get skip result like readT